Dashboard Configuration Variable Highlighting

aaditn opened the pull request:
Marking dashboard config variables if they've been changed from their uploaded value.

aaditn committed:
Dashboard Configuration Variable Highlighting

Marking dashboard config variables if they've been changed from them from their uploaded value

rbrott commented:
Thanks for the patch!

> Marking dashboard config variables if they've been changed from their uploaded value.

I take the uploaded value to mean the value that was in the file deployed to the bot. The client needs more information from the backend to provide that value and shouldn't assume that the first config received by the client contains the deployed values.

Also you should use `yarn format` to format `client` changes. It's also great for graphical changes if you can post a screenshot of the new UI.

mxtmx committed:
Merge branch 'acmerobotics:master' into markVariables

mxtmx committed:
Merge branch 'acmerobotics:master' into markVariables

mxtmx committed:
Merge branch 'acmerobotics:master' into markVariables

mxtmx committed:
Merge branch 'acmerobotics:master' into markVariables

mxtmx committed:
Revert "Dashboard Configuration Variable Highlighting"

This reverts commit 73439636dde47267e2a4d54f2f5fbaf5649a32b9.

mxtmx committed:
Add config baseline tracking and UI for modified variables

Introduces backend and frontend support for tracking baseline configuration values as deployed to the robot. Adds new message types and Redux state for baseline config, updates the UI to display and filter variables modified from the baseline, and provides visual indicators for such modifications. This helps users identify and manage configuration changes relative to the deployed baseline.

mxtmx commented:
This would also resolve #50.

mxtmx committed:
Rename filter icon

mxtmx committed:
Merge branch 'master' into markVariables

mxtmx commented:
Merge conflicts have been resolved

mxtmx committed:
Merge branch 'master' into markVariables

mxtmx committed:
Merge branch 'acmerobotics:master' into markVariables

mxtmx committed:
Use yellow dot and yarn format

mxtmx commented:
Sorry for the delay, just fixed!

mxtmx committed:
Merge branch 'acmerobotics:master' into markVariables

mxtmx committed:
Merge branch 'master' into markVariables

mxtmx committed:
Merge branch 'master' into markVariables

mxtmx committed:
UI indicator tweaks

mxtmx commented:
All those changes should be made! I also ran `yarn format`.
<img width="334" height="133" alt="Screenshot 2026-03-22 at 11 56 05 AM" src="https://github.com/user-attachments/assets/eb4f7762-07ae-4b1d-a39f-0df08c941d2a" />
<img width="334" height="133" alt="Screenshot 2026-03-22 at 11 56 00 AM" src="https://github.com/user-attachments/assets/153714cc-6f49-4665-ba5d-a6d0c7669fbe" />
